gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -237,12 +234,20 @@ def run(argv=None):
 
   pipeline_options = PipelineOptions(pipeline_args)
   with beam.Pipeline(options=pipeline_options) as p:
-    variants = (p
-                | 'ReadFromVcf' >> vcfio.ReadFromVcf(
-                    known_args.input_pattern,
-                    allow_malformed_records=known_args.allow_malformed_records)
-                | 'FilterVariants' >> filter_variants.FilterVariants(
-                    reference_names=known_args.reference_names))
+    variants = None
+    if known_args.optimize_for_large_inputs:
+      variants = (p
+                  | 'InputFilePattern' >> beam.Create(
+                      [known_args.input_pattern])
+                  | 'ReadAllFromVcf' >> vcfio.ReadAllFromVcf(
+                      allow_malformed_records=(
+                          known_args.allow_malformed_records)))
+    else:
+      variants = p | 'ReadFromVcf' >> vcfio.ReadFromVcf(
+          known_args.input_pattern,
+          allow_malformed_records=known_args.allow_malformed_records)
```

Comment:
Consider moving this logic to a private method `_read_variants`.

Reply:
Done.